### PR TITLE
ショッピングカート画面で商品が表示されるように修正しました

### PR DESCRIPTION
**Pull request by SekiMoemi**
ご確認お願いします。

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderToppingRepository.java [hunk 1]
```diff
@@ -17,17 +17,18 @@
...
-	 */
-	public void save(OrderTopping orderTopping) {
-		String sql = "INSERT INTO order_toppings (topping_id, order_item_id) VALUES (:topping_id, :order_item_id);";
-		SqlParameterSource param = new BeanPropertySqlParameterSource(orderTopping);
-		template.update(sql, param);
-	}
+  /**
+   * 注文トッピング情報を挿入します.
+   * 
+   * @param orderTopping 注文トッピング情報
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderToppingRepository.java:26]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 2]
```diff
@@ -30,187 +30,195 @@
...
+      beforeOrderId = nowOrderId;
+      beforeOederItemId = nowOrderItemId;
+    }
+    return orderList;
+  };
+
+  /**
+   * 注文情報を挿入します.
+   * 
+   * @param order 注文情報
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:144]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 3]
```diff
@@ -30,187 +30,195 @@
...
+   */
+  public Integer save(Order order) {
+    StringBuilder sql = new StringBuilder(
+        "insert into orders (user_id, status, total_price, order_date, destination_name, destination_email, destination_zipcode,");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:148]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 131).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 4]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append(
+        "destination_pref, destination_municipalities, destination_address, destination_tel, delivery_time, payment_method)");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:150]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 126).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 5]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append(
+        "VALUES (:userId, :status, :totalPrice, :orderDate, :destinationName, :destinationEmail, :destinationZipcode,");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:152]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 120).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 6]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append(
+        ":destinationPref, :destinationMunicipalities, :destinationAddress, :destinationTel, :deliveryTime, :paymentMethod)");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:154]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 126).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 7]
```diff
@@ -30,187 +30,195 @@
...
+    KeyHolder keyHolder = new GeneratedKeyHolder();
+    String[] keyColumnNames = {"id"};
+    template.update(sql.toString(), param, keyHolder, keyColumnNames);
+    return keyHolder.getKey().intValue();
+  }
+
+  /**
+   * 注文IDから注文情報を取得します.
+   * 
+   * @param id 注文ID
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:165]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 8]
```diff
@@ -30,187 +30,195 @@
...
+   * @return 注文情報
+   */
+  public Order load(Integer id) {
+    String sql =
+        "SELECT id, user_id, status, total_price, order_date, destination_name, destination_email, destination_zipcode, destination_pref, destination_municipalities, destination_address, destination_tel, delivery_time, payment_method FROM orders WHERE id=:id";
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:170]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 260).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 9]
```diff
@@ -30,187 +30,195 @@
...
+    SqlParameterSource param = new MapSqlParameterSource().addValue("id", id);
+    Order order = template.queryForObject(sql, param, ORDER_ROW_MAPPER);
+    return order;
+  }
+
+  /**
+   * ユーザのIDと状態から注文情報を検索します.
+   * 
+   * @param userId ユーザのID
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:179]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 10]
```diff
@@ -30,187 +30,195 @@
...
+   * @param status 状態
+   * @return Orderのリスト
+   */
+  public List<Order> findByUserIdAndStatus(Integer userId, Integer status) {
+    StringBuilder sql =
+        new StringBuilder("SELECT o.id AS o_id, user_id, status, total_price, order_date,");
+    sql.append("destination_name, destination_email,destination_zipcode, destination_pref,");
+    sql.append(
+        "destination_municipalities, destination_address,destination_tel, delivery_time, payment_method,");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:188]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 107).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 11]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append(
+        "o_item.id AS o_item_id, quantity, o_item.size AS size,item.id AS item_id, item.name AS item_name, description,");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:190]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 122).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 12]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append("item.price_m AS item_price_m, item.price_l AS item_price_l, image_path, deleted,");
+    sql.append("o_topping.id AS o_topping_id, topping.id AS topping_id,");
+    sql.append(
+        "topping.name AS topping_name, topping.price_m AS topping_price_m, topping.price_l AS topping_price_l ");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:194]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 113).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 13]
```diff
@@ -30,187 +30,195 @@
...
+    sql.append("FROM orders AS o ");
+    sql.append("LEFT OUTER JOIN order_items AS o_item ON o.id = o_item.order_id ");
+    sql.append("LEFT OUTER JOIN items AS item ON item.id = o_item.item_id ");
+    sql.append(
+        "LEFT OUTER JOIN order_toppings AS o_topping ON o_topping.order_item_id = o_item.id ");
+    sql.append(
+        "LEFT OUTER JOIN toppings AS topping ON topping.id = o_topping.topping_id ORDER BY order_date DESC, status, o.id DESC;");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:201]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 129).

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 14]
```diff
@@ -30,187 +30,195 @@
...
+    List<Order> orderList = template.query(sql.toString(), ORDER_RESULT_SET_EXTRACTOR);
+    return orderList;
+  }
+
+  /**
+   * 注文情報を更新します.
+   * 
+   * @param order 注文情報
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:209]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java [hunk 15]
```diff
@@ -30,187 +30,195 @@
...
+   */
+  public void update(Order order) {
+    StringBuilder sql = new StringBuilder(
+        "UPDATE orders SET user_id=:userId, status=:status, total_price=:totalPrice, ");
+    sql.append("order_date=:orderDate, destination_name=:destinationName, ");
+    sql.append("destination_email=:destinationEmail, destination_zipcode=:destinationZipcode, ");
+    sql.append(
+        "destination_pref=:destinationPref, destination_municipalities=:destinationMunicipalities, ");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/OrderRepository.java:217]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 102).

File: src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java [hunk 16]
```diff
@@ -9,51 +9,51 @@
...
-	public void setToppingIdList(List<Integer> toppingIdList) {
-		this.toppingIdList = toppingIdList;
-	}
-
-	@Override
-	public String toString() {
-		return "AddShoppingCartForm [itemId=" + itemId + ", quantity=" + quantity + ", size=" + size
-				+ ", toppingIdList=" + toppingIdList + "]";
-	}
+  /** 商品ID */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java:12]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java [hunk 17]
```diff
@@ -9,51 +9,51 @@
...
+  private String itemId;
+  /** 数量 */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java:14]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java [hunk 18]
```diff
@@ -9,51 +9,51 @@
...
+  private Integer quantity;
+  /** サイズ */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java:16]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java [hunk 19]
```diff
@@ -9,51 +9,51 @@
...
+  private String size;
+  /** トッピングIDのリスト */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/form/AddShoppingCartForm.java:18]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/jp/co/example/ecommerce_a/service/AddShoppingCartService.java [hunk 20]
```diff
@@ -22,55 +22,59 @@
...
-		for(OrderItem item : nowOrder.getOrderItemList()) {
-			totalPrice += item.getSubTotal();
-		}
-		nowOrder.setTotalPrice(totalPrice);
-		orderRepository.update(nowOrder);
-	}
+  /**
+   * ショッピングカートに商品を追加します.
+   * 
+   * @param form ショッピングカートへの追加時に利用するフォーム
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/AddShoppingCartService.java:35]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

File: src/main/java/jp/co/example/ecommerce_a/service/AddShoppingCartService.java [hunk 21]
```diff
@@ -22,55 +22,59 @@
...
+      order.setUserId(userId);
+      order.setStatus(0);
+      order.setTotalPrice(0);
+      orderId = orderRepository.save(order);
+      order.setId(orderId);
+    } else {
+      orderId = orders.get(0).getId();
+      order = orders.get(0);
+    }
+    int orderItemId = 0;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/AddShoppingCartService.java:53]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'orderItemId' declaration and its first usage is 5, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

**Review by yamada-kensyu — APPROVED** [review]
承認します

**Pull request merged**